src/nbl/video/CVKLogicalDevice.h

```diff
@@ -124,51 +327,472 @@ class CVKLogicalDevice final : public ILogicalDevice
+        if (cpushader->containsGLSL())
+        {
+            return core::make_smart_refctd_ptr<CVulkanShader>(
+                core::smart_refctd_ptr<CVKLogicalDevice>(this), std::move(clone),
+                IGPUShader::buffer_contains_glsl);
+        }
+        else
+        {
+            return core::make_smart_refctd_ptr<CVulkanShader>(core::smart_refctd_ptr<CVKLogicalDevice>(this),
+                std::move(clone));
+        }
+    }
+
+    core::smart_refctd_ptr<IGPUImage> createGPUImage(asset::IImage::SCreationParams&& params) override
+    {
+        VkImageCreateInfo vk_createInfo = { VK_STRUCTURE_TYPE_IMAGE_CREATE_INFO };
+        vk_createInfo.pNext = nullptr; // there are a lot of extensions
+        vk_createInfo.flags = static_cast<VkImageCreateFlags>(params.flags);
+        vk_createInfo.imageType = static_cast<VkImageType>(params.type);
+        vk_createInfo.format = getVkFormatFromFormat(params.format);
+        vk_createInfo.extent = { params.extent.width, params.extent.height, params.extent.depth };
+        vk_createInfo.mipLevels = params.mipLevels;
+        vk_createInfo.arrayLayers = params.arrayLayers;
+        vk_createInfo.samples = static_cast<VkSampleCountFlagBits>(params.samples);
+        vk_createInfo.tiling = static_cast<VkImageTiling>(params.tiling);
+        vk_createInfo.usage = static_cast<VkImageUsageFlags>(params.usage);
+        vk_createInfo.sharingMode = static_cast<VkSharingMode>(params.sharingMode);
+        vk_createInfo.queueFamilyIndexCount = params.queueFamilyIndices->size();
+        vk_createInfo.pQueueFamilyIndices = params.queueFamilyIndices->data();
+        vk_createInfo.initialLayout = static_cast<VkImageLayout>(params.initialLayout);
+
+        VkImage vk_image;
+        if (vkCreateImage(m_vkdev, &vk_createInfo, nullptr, &vk_image) == VK_SUCCESS)
+        {
+            return core::make_smart_refctd_ptr<CVulkanImage>(core::smart_refctd_ptr<CVKLogicalDevice>(this),
+                std::move(params), vk_image);
+        }
+        else
+        {
+            return nullptr;
+        }
+    }
+
+    bool bindImageMemory(uint32_t bindInfoCount, const SBindImageMemoryInfo* pBindInfos) override
+    {
+        for (uint32_t i = 0u; i < bindInfoCount; ++i)
+        {
+            if (pBindInfos[i].image->getAPIType() != EAT_VULKAN)
+                continue;
+            VkImage vk_image = static_cast<const CVulkanImage*>(pBindInfos[i].image)->getInternalObject();
+
+            // if (pBindInfos[i].memory->getAPIType() != EAT_VULKAN)
+            //     continue;
+            VkDeviceMemory vk_deviceMemory = static_cast<const CVulkanMemoryAllocation*>(pBindInfos[i].memory)->getInternalObject();
+
+            if (vkBindImageMemory(m_vkdev, vk_image, vk_deviceMemory,
+                static_cast<VkDeviceSize>(pBindInfos[i].offset)) != VK_SUCCESS)
+            {
+                return false;
+            }
```

Comment:
again, dont quit on first fail

Also you've forgotten to set the memory binding smart pointer

Reply:
Resolved

Reply:
@Erfan-Ahmadi same fail as with the `bindBufferMemory` the `setMemoryAndOFfset` should only be called when the vulkan call succeeds